### PR TITLE
integrate isDeflate and isGzip

### PR DESCRIPTION
**Pull request by Uzlopak**
integrate isDeflate and isGzip
remove corresponding packages from dependencies
wrote unit tests for the above functions

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -1,7 +1,28 @@
 'use strict'
 
+function isDeflate (buffer) {
```
Comment (Eomm) [inline lib/utils.js:3]: what is the scope of this change?
Why should we copy-paste those modules into this one? (at least credits should be added)

Comment (Fdawgs) [reply lib/utils.js:3]: One less dependency that could be potentially compromised?

**Review by kibertoad — APPROVED** [review]

File: lib/utils.js (hunk 1 above)
Comment (Uzlopak) [reply lib/utils.js:4]: isDeflate is now spec compliant and definetly a complete new product. I benched the performance of isDeflate an it has still the same performance with the replaced one.

I also added references to the corresponding RFCs to explain the magic numbers.

**Review by Fdawgs — APPROVED** [review]

**Comment by Uzlopak** [comment]
@kibertoad 
@Fdawgs 

I have to ask you to again review it. On a smoking break I realized that i can do the CMF header check in isDeflate with bit operations. 

before:
```
valid x 223,947,412 ops/sec ±0.70% (90 runs sampled)
invalid x 831,593,213 ops/sec ±0.88% (93 runs sampled)
```

after:
```
valid x 1,027,199,609 ops/sec ±0.74% (89 runs sampled)
invalid x 1,038,269,211 ops/sec ±0.94% (91 runs sampled)
```

```
'use strict'

const { Suite } = require('benchmark')
const isDeflate = require('./lib/utils').isDeflate

const suite = new Suite()

const fakeDeflate = Buffer.from([0x78, 0x9d])
const fakeGzip = Buffer.from(Buffer.from([0x1f, 0x8b, 0x08]))

suite.add('valid', function () {
  isDeflate(fakeDeflate)
}).add('invalid', function () {
  isDeflate(fakeGzip)
})
suite
  .on('cycle', function (event) {
    console.log(String(event.target))
  })
  .on('complete', function () {
    console.log('Fastest is ' + this.filter('fastest').map('name'))
    console.log('Slowest is ' + this.filter('slowest').map('name'))
  })
  .run({ async: true })

```

**Review by Fdawgs — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**